articles/5_sass_functions.md

```diff
@@ -0,0 +1,337 @@
+
+```scss
+@function get-region-color($state) {
+  $region-colors: red, blue, green;
+
+  @for $i from 1 through 3 {
+    $region: nth($regions, $i);
+
+    @if index($region, $state) {
+      @return nth($region-colors, $i);
+    }
+  }
+
+  @return black;
+}
+```
+
+Then to get the color, the function needs to be called
+
+```scss
+#NY {
+  color: get-region-color("NY");
+}
+```
+
+
+Here is an identical [codepen demo](http://codepen.io/raemadeline/pen/Krybaq) using the function instead of the mixin.
+
+## In-Depth Example: Refactoring Z-Indexes
+
+Last month, I refactored the way we do `z-index`es at Addepar to leverage the powers of SASS. This refactor encompassed all of the concepts I discussed above, so I thought it would make a good case study.
+
+### What is a `z-index`?
+
+The `z-index` of an element refers to its positioning in the z-axis. If you think of the x- and y- axes as the left-right / up-down directions on the screen, the z-axis is the direction going into and out of the screen. Another way of thinking about it is the way the element layers above or below other elements on the page.
+
+![z-index](../images/z-index.gif)
+
+*Image courtesy of http://www.annyhe.com/*
+
+Something to note that is not directly related to the larger topic of this post, but is related to `z-index`es: When applying a `z-index` to elements nested within each other, the child element's maximum place in the layering order is that of its parent, no matter how high the child element's `z-index` is.
+
+For a visual representation of how that looks (Assume that boxes 2 and 3 are children of box 1):
+
+![stacking order](../images/stacking1.png)
+
+*Image courtesy of webdesign.tutsplus.com*
+
+For this reason, its important to avoid putting `z-index`es on elements that don't absolutely need them, as it can lead to some unintended consequences.
+
+### Why did I need to refactor them?
+
+Initially, all of our `z-index`es were stored as individual variables in a file that looked like this:
+
+```scss
+$zindex-navbar:                  1000;		
+$zindex-dropdown:                1000;		
+$zindex-popover:                 1010;		
+$zindex-tooltip:                 1030;		
+$zindex-navbar-fixed:            1030;		
+$zindex-modal-background:        1040;		
+$zindex-modal:                   1050;
+...
+```
+
+While these initial values are fairly well organized, this quickly fell into chaos as every new component or one-off change to a `z-index` just led to creating new variables with magic numbers, or worse, just putting the magic number directly into the declaration without explanation for why that element had to be layered that way.
+
+### How did I fix it?
+
+Now, our `z-index`es are organized in a nested map structure based on where they appear in the app.
+
+```scss
+$z-layers: (
```

Comment:
I feel like the data in this map stills feels magical. It might be worth explaining more about how you went from the z-index variables above to this data structure.